core/src/client/mod.rs

```diff
@@ -51,6 +52,31 @@ use serde::de::DeserializeOwned;
 use serde_json::Value as JsonValue;
 use tokio::sync::{mpsc, oneshot};
 
+/// Shared state whether a subscription has lagged or not.
+#[derive(Debug, Clone)]
+pub(crate) struct SubscriptionLagged(Arc<Mutex<bool>>);
```

Comment:
Any reason not to use `Arc<AtomicBool>` instead? :)

Reply:
Actually I endedu pthinking more about this and the mutex does offer stronger guarantees I think re ordering! It's probably the safe option regardless without thinking very carefully about it!